ModelBundle/Repository/NodeRepository.php

```diff
@@ -729,6 +729,29 @@ public function findLastPublished()
 
     /**
      * @param string $nodeType
+     * @param int    $skip
+     * @param int    $limit
+     *
+     * @return array
+     */
+    public function findAllCurrentlyPublishedByTypeWithSkipAndLimit($nodeType, $skip, $limit)
+    {
+        $qa = $this->createAggregationQuery();
+        $qa->match(
+            array(
+                'nodeType' => $nodeType,
+                'currentlyPublished' => true,
+                'deleted' => false
+            )
+        );
+        $qa->skip($skip);
+        $qa->limit($limit);
```

Comment:
@alavieille : I think you should add an order to be sure that items selected in the range 1-20 are not present again in the range 21-40 for instance